### PR TITLE
feat: support custom environment variables for arbitrary CLIs

### PR DESCRIPTION
**Pull request by yj-yan**
Please read [CONTRIBUTING.md](CONTRIBUTING.md) for additional information on contributing to this repository!

## What this PR does / why we need it
Add block for additional env variables in goreleaser.

## Jira ID

[DT-4161]

## Notes for your reviewers

[DT-4161]: https://outreach-io.atlassian.net/browse/DT-4161?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by malept — CHANGES_REQUESTED** [review]
I think we shouldn't be hardcoding these kinds of things in `stencil-golang` since it's a public repository. I think the better way to solve this problem is to add a Stencil block here.

**Comment by yj-yan** [comment]
I updated the indent for env block, thanks for the help. @malept

File: templates/.goreleaser.yml.tpl [hunk 1]
```diff
@@ -29,7 +29,12 @@ builds:
    {{- end }}
   env:
   - CGO_ENABLED={{ stencil.ApplyTemplate "cgoEnabled" | trim }}
-{{- end }}
+  {{- $blockName := (printf "%vAdditionalEnv" $cmdName) }}
+  ## <<Stencil::Block({{ $blockName }})>>
+  {{- $env := (file.Block $blockName ) | trim }}
+{{ $env | indent 2}}
```
Comment (malept) [inline templates/.goreleaser.yml.tpl:35]: suggestion: does it still work if you merge this into one line?
```suggestion
{{ (file.Block $blockName) | trim | indent 2 }}
```

Comment (yj-yan) [reply templates/.goreleaser.yml.tpl:35]: > (file.Block $blockName)

Good catch. $env was just initially created to check if the block is empty. Applied the change.

**Review by malept — APPROVED** [review]

**Pull request merged**